File: src/decoders/restaurant-kind.ts
import { RestaurantKind, UnknownEnumValue } from "~/models";

export function decodeRestaurantKind (value: any): RestaurantKind {
  switch (value) {
    case "Cafétéria":
      return RestaurantKind.CAFETERIA;
    case "Restaurant":
      return RestaurantKind.RESTAURANT;
    case "Restaurant agréé":
      return RestaurantKind.APPROVED_RESTAURANT;
    case "Restaurant géré":
      return RestaurantKind.MANAGED_RESTAURANT;
    case "Coffee Corner":
      return RestaurantKind.COFFEE_CORNER;
    case "Brasserie":
      return RestaurantKind.BREWERY;
    case "Foodtruck":
      return RestaurantKind.FOOD_TRUCK;
    case "Restaurant administratif":
      return RestaurantKind.ADMINISTRATIVE_RESTAURANT;
    case "Libre-service":
      return RestaurantKind.SELF_SERVICE;
    case "Kiosque":
      return RestaurantKind.KIOSK;
    case "Pizzéria":
      return RestaurantKind.PIZZERIA;
    case "épicerie":
      return RestaurantKind.GROCERY_STORE;
    case "Triporteur":
      return RestaurantKind.SCOOTER;
    case "crous and go":
      return RestaurantKind.CROUS_AND_GO;
    case "Sandwicherie":
      return RestaurantKind.SANDWICH_SHOP;
    default:
      throw new UnknownEnumValue("RestaurantKind", value);
  }
};

File: library/src/commonMain/kotlin/decoders/restaurant-kind.kt
package decoders

import models.RestaurantKind

fun decodeRestaurantKind (kind: String): RestaurantKind {
  return when (kind) {
    "Cafétéria" -> RestaurantKind.CAFETERIA
    "Restaurant" -> RestaurantKind.RESTAURANT
    "Restaurant agréé" -> RestaurantKind.APPROVED_RESTAURANT
    "Restaurant géré" -> RestaurantKind.MANAGED_RESTAURANT
    "Coffee Corner" -> RestaurantKind.COFFEE_CORNER
    "Brasserie" -> RestaurantKind.BREWERY
    "Foodtruck" -> RestaurantKind.FOOD_TRUCK
    "Restaurant administratif" -> RestaurantKind.ADMINISTRATIVE_RESTAURANT
    "Libre-service" -> RestaurantKind.SELF_SERVICE
    "Kiosque" -> RestaurantKind.KIOSK
    "Pizzéria" -> RestaurantKind.PIZZERIA
    "épicerie" -> RestaurantKind.GROCERY_STORE
    "Triporteur" -> RestaurantKind.SCOOTER
    "crous and go" -> RestaurantKind.CROUS_AND_GO
    "Sandwicherie" -> RestaurantKind.SANDWICH_SHOP
    else -> throw Exception("Unknown restaurant kind: $kind")
  }
}
